### DISCUSSION TITLE
Loading OstrichRL using MuJoCo Unity Plug-in

### DISCUSSION THREAD

**Discussion by desaibhargav** [google-deepmind/mujoco#2169] [Asking for Help]
### Intro

Hi!

I am a PhD student at Indiana University Bloomington, I use MuJoCo and the Unity plugin for my research on emergent behaviors.

### My setup

MuJoCo 3.2.3
Unity-Plugin 3.2.3
macOS Sequoia 15.0.1 (24A348), Apple Silicon 

### My question

Thank you for this plugin, it is really useful as my other research directions use Unity and I wanted to integrate MuJoCo into my workflows without any major refactors. 

I was experimenting with the plugin, and I noticed that some XMLs load perfectly, while many don't (even the ones in the [Menagerie](https://github.com/google-deepmind/mujoco_menagerie) project. 

For my specific use case I was interested in loading the [OstrichRL](https://github.com/vittorione94/ostrichrl) model, which loads fine in the MuJoCo macOS Application, as well as when using Python, but fails with the plug-in. 

I can provide a more comprehensive list of models that fail if that helps the development team. 

### Minimal model and/or code that explain my question

The model is too big for the 25MB limit here, but can be found [here](https://github.com/vittorione94/ostrichrl)

The XML file is at /ostrichrl/assets/models/ostrich

### Confirmations

- [x] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
Perhaps @Balint-H and/or @vittorione94 can comment on this

**Reply by Balint-H** [reply]
On it, will give feedback this week.

**Comment by Balint-H** [comment]
@desaibhargav Many apologies for not getting back to you. I expect you have stopped working on the topic, but in any case I finally looked into it. 

At least on v3.2.4 and later the ostrich model can be imported to Unity and is able to be simulated- the only issue seems to be needing to manually set the actuator type to muscle which is a bug I'll fix, thanks for bringing it to my attention.

The materials of geoms do need manual configuring as well, but that is common for most models in the Unity plugin. Lastly, due one side of the legs being mirrored, they are rendered as internal faces in Unity only. This is not a bug, but a common optimisation Unity makes to only render one side of meshes. The affected meshes could be batch "turned inside out" with software like Blender. 

Please let me know if you are still interested and have more questions!
